### ISSUE TITLE
Claimed twice after settlement

### ISSUE THREAD

**Issue by FedererKK** [vyper-protocol/vyper-otc-ui#249]
**Describe the bug**
After settling the [following contract](3tmR65f7sULhZdjM2fw9XAFmPdfj7MwTygWhiFaEvdQj) I was able to call "Claim" twice.

First claim I redeemed the right amount (https://solscan.io/tx/5xvReEepzqamD2w87nCPUUfdXU5XnHqQqTCTUBApSg4dHG2hAF6jCkzKezvLxaZLLoRrZdMSQyAPh97uwukPYjgx)

Second claim I redeemed zero (https://solscan.io/tx/34phi5zC7RvmoodUrSdb5mrNuBunB7GXkMFppxZdMPEgzqojGM7iy8QxSQpuQok2EB5VK3vNymc5W2Q3rgHcMxMm)

**Expected behavior**
No second claim should be possible.

**Comment by thegreat-vanderlinde** [comment]
This is true. On the smart contract level this is currently possible, during claim we simply transfer assets from the settled account to the beneficiary account, with no checking on the amount. So in case of multiple claims the first one will transfer all assets and the following will transfer the token amount left, zero.

https://github.com/vyper-protocol/vyper-otc/blob/870d7ed3b3af56ea7a541c8dca45a2624b196905/programs/vyper-otc/src/instructions/claim.rs#L73-L81

We can fix this on the UI for now.

**Comment by thegreat-vanderlinde** [comment]
@Shikhar15606 that's smart but unfortunately the check is already in place:
https://github.com/vyper-protocol/vyper-otc-ui/blob/785767d8ad17ee6574d98ac04a37b8da06f78905/src/models/ChainOtcState.ts#L96-L104

the problem here is that the UI doesn't update _fastly_, the chain query to fetch data get old data also after the tx submit. We have to include a listener in the button as we did for the deposit here:
https://github.com/vyper-protocol/vyper-otc-ui/blob/785767d8ad17ee6574d98ac04a37b8da06f78905/src/components/organisms/actionButtons/DepositButton.tsx#L29-L72

**Comment by Shikhar15606** [comment]
@thegreat-vanderlinde I am thinking of adding a ```onAccountChange``` listener on the token accounts: ```otcSeniorReserveTokenAccount``` or ```otcJuniorReserveTokenAccount```

According to my understanding of the ```onAccountChange``` method => It should fire when the tokens are transferred from it to the beneficiary account. 

But when I looked at the ```ChainOtcState Class``` I didn't get the token accounts ```otcSeniorReserveTokenAccount``` and ```otcJuniorReserveTokenAccount``` in it. So shall I add them to the class ??
